### PR TITLE
Update grpc to 1.0.1 and protobuf to 3.0.0

### PR DESCRIPTION
**Pull request by lukecwik**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by lukecwik** [comment]
Old mvn dependency tree
http://www.heypasteit.com/clip/2XXN
New mvn dependency tree
http://www.heypasteit.com/clip/2XXR
R: @peihe

File: sdks/java/core/src/test/java/org/apache/beam/sdk/util/PubsubGrpcClientTest.java [hunk 1]
```diff
@@ -73,31 +79,24 @@
   private static final String ACK_ID = "testAckId";
 
   @Before
-  public void setup() throws IOException {
-    mockChannel = Mockito.mock(ManagedChannel.class);
+  public void setup() {
+    channelName = PubsubGrpcClientTest.class.getName()
```
Comment (peihe) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/util/PubsubGrpcClientTest.java]: nit: use String.format?

File: sdks/java/core/pom.xml [hunk 2]
```diff
@@ -337,22 +337,7 @@
...
-          <artifactId>guava-jdk5</artifactId>
-        </exclusion>
-        <!-- grpc-all does not obey IWYU, so we need to exclude from compile
-          scope and depend on it at runtime. -->
-        <exclusion>
-          <groupId>io.grpc</groupId>
-          <artifactId>grpc-all</artifactId>
-        </exclusion>
-      </exclusions>
+      <artifactId>grpc-google-pubsub-v1</artifactId>
```
Comment (lukecwik) [inline sdks/java/core/pom.xml]: pubsub team renamed the artifact that they publish to maven
note that pubsub proto/service definitions that we use didn't change

File: pom.xml [hunk 3]
```diff
@@ -103,19 +103,19 @@
...
     <bigquery.version>v2-rev295-1.22.0</bigquery.version>
-    <pubsubgrpc.version>0.0.2</pubsubgrpc.version>
+    <pubsubgrpc.version>0.1.0</pubsubgrpc.version>
     <clouddebugger.version>v2-rev8-1.22.0</clouddebugger.version>
     <dataflow.version>v1b3-rev36-1.22.0</dataflow.version>
     <dataflow.proto.version>0.5.160222</dataflow.proto.version>
-    <datastore.client.version>1.1.0</datastore.client.version>
-    <datastore.proto.version>1.0.1</datastore.proto.version>
+    <datastore.client.version>1.2.0</datastore.client.version>
+    <datastore.proto.version>1.2.0</datastore.proto.version>
```
Comment (lukecwik) [inline pom.xml]: its for the grpc/protobuf changes
those libraries depend on grpc 1.0.0+ and protobuf 3.0.0+

File: pom.xml [hunk 4]
```diff
@@ -321,11 +321,35 @@
...
+      <dependency>
+        <groupId>io.grpc</groupId>
+        <artifactId>grpc-protobuf-lite</artifactId>
+        <version>${grpc.version}</version>
+      </dependency>
+
+      <dependency>
+        <groupId>com.google.protobuf</groupId>
+        <artifactId>protobuf-lite</artifactId>
+        <version>3.0.1</version>
```
Comment (lukecwik) [inline pom.xml]: No, unfortunately grpc pulls in 3.0.1 and not 3.0.0

File: sdks/java/core/src/test/java/org/apache/beam/sdk/util/PubsubGrpcClientTest.java (hunk 1 above)
Comment (lukecwik) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/util/PubsubGrpcClientTest.java]: Done

**Comment by peihe** [comment]
LGTM

**Pull request closed without merging**